src/matcha_ml/cli/provision.py

```diff
@@ -75,6 +75,20 @@ def provision_resources(
     """
     remote_state_manager = RemoteStateManager()
 
+    if remote_state_manager.is_state_stale():
+        print(
```

Comment:
Nit: This feels like an important warning message to the user. Could we wrap this in `build_warning_status` from `ui.status_message_builders`?